tests/crypto/src/common_test.h

```diff
@@ -0,0 +1,497 @@
+	const char                      *p_expected_output;             /**< Pointer to expected hmac digest in hex string format. */
+} test_vector_hmac_t;
+
+/**@brief hkdf test vector information.
+ */
+typedef const struct {
+	const u32_t digest_type;                                        /**< Digest type of hkdf operation. */
+	const int expected_err_code;                                    /**< Expected error code from hkdf operation. */
+	const int expected_err_code_expand;                             /**< Expected error code from hkdf expand operation. */
+	const uint8_t expected_result;                                  /**< Expected result of hkdf operation. */
+	const uint8_t expected_result_expand;                           /**< Expected result of hkdf expand operation. */
+	const char                      *p_test_vector_name;            /**< Pointer to hkdf test vector name. */
+	const char                      *p_ikm;                         /**< Pointer to hkdf Input Key Material in hex string format. */
+	const char                      *p_okm;                         /**< Pointer to hkdf Output Key Material in hex string format. */
+	const char                      *p_prk;                         /**< Pointer to hkdf PseudoRandom Key in hex string format. */
+	const char                      *p_salt;                        /**< Pointer to hkdf salt in hex string format. */
+	const char                      *p_info;                        /**< Pointer to hkdf optional application specific information in hex string format. */
+} test_vector_hkdf_t;
+
+const char *get_vector_name(const test_case_t *tc, uint32_t v);
+uint32_t get_vector_count(const test_case_t *tc);
+
+
+
+/**@brief Function for initializing hardware acceleration.
+ *
+ * @details Should only be called once.
+ *
+ * @return								0 on success, else an error code.
+ */
+int init_cc310(void);
+
+/**@brief Function for initializing counter mode deterministic random byte generator.
+ *
+ * @details Should only be called once.
+ *
+ * @param[in]		p_optional_seed		Pointer to a seed. Can be NULL. In that case uses internal seed.
+ * @param[in]      	len					Length of seed in bytes. Ignored if seed is NULL.
+ *
+ * @return								0 on success, else an error code.
+ */
+#if defined(CONFIG_MBEDTLS_CTR_DRBG_ENABLED)
+int init_ctr_drbg(const unsigned char *p_optional_seed, size_t len);
+#endif
+
+/**@brief Function for setting LED_1 to indicate a start of time measurement.
+ *
+ */
+void start_time_measurement(void);
+
+/**@brief Function for clearing LED_1 to indicate a stop of time measurement.
+ *
+ */
+void stop_time_measurement(void);
+
+/**@brief Macro(s) for decorating test vector names with file and line information.
+ *
+ */
+#define _TV_DECORATE2(file, line, name)     name " -- " "[" file ":" #line "]"
+#define _TV_DECORATE1(file, line, name)     _TV_DECORATE2(file, line, name)
+#define TV_NAME(name)                       _TV_DECORATE1(__FILE__, __LINE__, name)
```

Comment:
Does

name " -- [" __FILE__ ":" __LINE__ "]"

not work?

or at least reducing the levels of indirection from 2 to 1 ?

Reply:
As far as I can tell, two additional macros are necessary _after_ the macro which contains `__LINE__`.
See e.g. this answer: https://stackoverflow.com/a/2670919
Therefore I don't see any way to make this cleaner.
There might be some preprocessor magic that would make this less verbose.

Reply:
Ah, was not aware __LINE__ was an integer, in that case we can use this magic:

`#define TV_NAME(name) name " -- [" __FILE__ ":" STRINGIFY(__LINE__) "]"
`

STRINGIFY does the two levels of indirection. It comes from zephyr/include/toolchain/common.h and will be available already I believe.

Reply:
Changed to the suggested line in adc45bd